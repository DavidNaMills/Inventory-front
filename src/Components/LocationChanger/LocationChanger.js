import React from 'react';
import classes from './locationChanger.module.scss';
import { useSelector } from 'react-redux';
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';
import PlaceIcon from '@material-ui/icons/Place';


const LocationChanger = () => {
    const { staff, types } = useSelector(state => state);
    const { changeCurrentLocationDispatch } = useDispatchHook();
    return (
        (staff.token && Object.keys(types.locations).length > 0)
        && <div className={classes.locationChanger}>
            <span className={classes.locationChanger__bold}><PlaceIcon/> </span>


            {(staff.staff.baseId.length > 1)
                ?
                <select onChange={(e) => changeCurrentLocationDispatch(e.target.value)} className={classes.locationChanger__selectStyle}>
                    {staff.staff.baseId.map(x => <option key={x} value={x}>{types.locations[x].displayValue}</option>)}
                </select>
                : <p>{types.locations[staff.staff.baseId[0]].displayValue}</p>
            }
        </div>

    )
}
export default React.memo(LocationChanger);