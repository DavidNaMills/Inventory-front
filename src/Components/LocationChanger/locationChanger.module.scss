.locationChanger{
    display: flex;
    align-items: center;

    &__selectStyle{
        border-radius: 10px;
        width: 200px;
        padding: 3px;
    }

    &__bold{
        font-weight: bold;
        margin-right: 15px;
    }
}