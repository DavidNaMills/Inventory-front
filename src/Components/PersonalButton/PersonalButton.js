import React from 'react';
import classes from './personalButton.module.scss';
import { NavLink } from 'react-router-dom';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';



const PersonalButton = () => (
    <div>
        <NavLink
            to='/personal'
            activeClassName={classes.personalButton__active}
            className={classes.personalButton}
        >
            <AccountCircleIcon size={50} />
        </NavLink>
    </div>
)

export default React.memo(PersonalButton);