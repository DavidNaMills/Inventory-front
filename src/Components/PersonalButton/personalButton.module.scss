@import '../../styles/variables/colors.module.scss';

.personalButton{
    display: flex;
    align-items: baseline;
    margin-right: 10px;
    color: $nav-color-inactive;

    &__active{
        color: $title-color;
        
    }

    &:hover{
        color: $nav-hover;
    }

    span{
        padding-right: 10px;
    }
}