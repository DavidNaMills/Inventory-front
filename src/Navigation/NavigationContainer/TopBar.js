import React from 'react';
import LocationChanger from '../../Components/LocationChanger/LocationChanger';
import PersonalButton from '../../Components/PersonalButton/PersonalButton';


const style = {
    display: 'flex',
    justifyContent: 'space-between',
    backgroundColor: 'rgb(193, 224, 207)',
    padding: '10px',
    borderBottom: 'solid rgb(132, 184, 155) 3px'
}


const TopBar = ({ name }) => (
    <div style={style}>
        <LocationChanger />
        <PersonalButton name={name} />
    </div>
)

export default TopBar;