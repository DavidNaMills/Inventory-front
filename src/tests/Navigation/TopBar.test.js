import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import TopBar from '../../Navigation/NavigationContainer/TopBar';
import LocationChanger from '../../Components/LocationChanger/LocationChanger';
import PersonalButton from '../../Components/PersonalButton/PersonalButton';


jest.mock('react-redux');
import { useSelector } from 'react-redux';

const mockStaff = (token) => ({
    token: token,
    staff: {
        name: 'David',
        baseId: ['123', '456123']
    },
    config: {
        routes: [
            { to: '/dashboard', label: 'nav_home' },
            { to: '/products', label: 'nav_prod_man' },
            { to: '/productTypes', label: 'nav_prod_ty' },
            { to: '/locations', label: 'nav_location' },
            { to: '/staff', label: 'nav_staff' },
            { to: '/test', label: 'nav_staff' }
        ]
    }

})

let wrapper;

afterEach(() => {
    jest.clearAllMocks();
});


beforeEach(() => {
    useSelector.mockImplementation(() => ({
        staff: { ...mockStaff('123123123') },
        types: {
            locations: {
                '123': { displayValue: 'home' },
                '456123': { displayValue: 'home2' },
            }
        }
    }));

    wrapper = mount(<BrowserRouter><TopBar name={'David'} /></BrowserRouter>);
})


describe('TopBar test suite', () => {
    
    it('matches snapshot', () => {
        const comp = renderer.create(<BrowserRouter><TopBar name={'David'} /></BrowserRouter>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy()
    });

    it('renders <LocationChanger/> and <PersonalButton/> components', ()=>{
        expect(wrapper.find(LocationChanger).length).toBe(1);
        expect(wrapper.find(PersonalButton).length).toBe(1);
    });
});