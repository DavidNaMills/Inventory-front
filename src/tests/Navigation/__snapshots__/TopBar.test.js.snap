// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar test suite matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgb(193, 224, 207)",
      "borderBottom": "solid rgb(132, 184, 155) 3px",
      "display": "flex",
      "justifyContent": "space-between",
      "padding": "10px",
    }
  }
>
  <div
    className="locationChanger"
  >
    <span
      className="locationChanger__bold"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
       
    </span>
    <select
      className="locationChanger__selectStyle"
      onChange={[Function]}
    >
      <option
        value="123"
      >
        home
      </option>
      <option
        value="456123"
      >
        home2
      </option>
    </select>
  </div>
  <div>
    <a
      aria-current={null}
      className="personalButton"
      href="/personal"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        size={50}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
        />
      </svg>
    </a>
  </div>
</div>
`;
